### PR TITLE
build updates

### PR DESCRIPTION
**Pull request by burtbeckwith**

### CODE REVIEW

File: tests/tasks-sqs/src/main/java/example/TasksPublisher.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ public void sendMessages() {
         try {
             int id = tasksIds.incrementAndGet();
             LOGGER.info("Publishing a task with id: " + id);
-            tasksProducer.send(new Task(id), "" + id);
+            tasksProducer.send(new Task(id));
```
Comment (graemerocher) [inline tests/tasks-sqs/src/main/java/example/TasksPublisher.java:26]: Why was this change necessary?

Comment (burtbeckwith) [reply tests/tasks-sqs/src/main/java/example/TasksPublisher.java]: SQS messages were failing complaining about the message header id but the error message wasn't helpful. turns out it was because that header is only valid with FIFO queues but the queue was a regular queue with a FIFO name. I updated queue creation and restored the header

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
     id 'groovy-gradle-plugin'
 }
+
+repositories {
+    gradlePluginPortal()
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'io.micronaut.gradle:micronaut-gradle-plugin:3.1.0'
```
Comment (melix) [inline buildSrc/build.gradle:11]: You can use `3.1.1`.

File: buildSrc/src/main/groovy/io.micronaut.build.internal.jms-examples.gradle [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    id 'io.micronaut.application'
+}
+
+micronaut {
+    version project.ext.micronautVersion
+    runtime 'netty'
+}
+
+dependencies {
+    testCompileOnly 'com.google.code.findbugs:jsr305' // for "warning: unknown enum constant When.MAYBE"
```
Comment (melix) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.jms-examples.gradle:12]: Any reason you don't use the catalog for this one?

You could declare a library without version like this:

```toml
[libraries]
findbugs = { module = "com.google.code.findbugs:jsr305", version = "" }
```

File: buildSrc/src/main/groovy/io.micronaut.build.internal.jms-tests.gradle [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+micronaut {
+    version project.ext.micronautVersion
+    runtime 'netty'
+    testRuntime 'spock'
+}
+
+dependencies {
+    testImplementation "io.micronaut.test:micronaut-test-core:$micronautTestVersion"
```
Comment (melix) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.jms-tests.gradle:18]: Nothing to do yet but as we move modules to use version catalogs we'll have to support a couple of things:

- versions like `micronautTestVersion` will need to be declared in the catalog, not `gradle.properties`
- the BOM update workflow will need to support updating from the catalog, currently it only supports `gradle.properties`

File: docs-examples/example-groovy/build.gradle [hunk 5]
```diff
@@ -9,6 +10,6 @@ micronaut {
 mainClassName = 'io.micronaut.jms.docs.Application'
 
 project.afterEvaluate {
-    nativeImage.enabled = false
+    nativeCompile.enabled = false
```
Comment (melix) [inline docs-examples/example-groovy/build.gradle:13]: :x: Avoid using afterEvaluate and referencing tasks directly as it eagerly configures them. Instead you can write:

```
tasks.withType(org.graalvm.buildtools.gradle.tasks.BuildNativeImageTask).configureEach {
    enabled = false
}
```

**Review by melix — CHANGES_REQUESTED** [review]
Good to see some more build cleanups! I left a few comments, but nothing critical!

**Review by melix — APPROVED** [review]
LGTM! Good to see that the minimal application plugin already helps!
